wms/models/datasets/ugrid.py

```diff
@@ -284,7 +281,29 @@ def getmap(self, layer, request):
                     return self.empty_response(layer, request)
 
                 if request.GET['image_type'] == 'filledcontours':
-                    return mpl_handler.tricontourf_response(tri_subset, data, request)
+                    if np.isnan(np.sum(data)):
+                        data_mask = np.logical_not(np.isnan(data))  # mask for non-NaN data elements
```

Comment:
Python method calls are a known bottleneck.  The difference might seem small now, but with subsequent calls this might make a difference.  I remember an old project of mine where changing the calls change from `np.deg2rad` to `deg2rad` make it 10% faster!  (To be fair there was a lot of those calls in a gigantic loop.)

And if this change makes the code clearer it is a win-win :wink: